### ISSUE TITLE
Delly crashes inside the docker with several complaints

### ISSUE THREAD

**Issue by mdperry** [ICGC-TCGA-PanCancer/pcawg_delly_workflow#9]
Hi,

Now that I have resolved the issues with the version of cwltool, and the lack of storage space in my root directory, the dockstore command launches the delly workflow for me, however after about 73 minutes is crashes (and once gain the /outputs directory on the host gets deleted somehow.  Here are three excerpts from the --debug log file that contain clear error messages:
1. The first inkling of trouble in the log file:
   Job Name: s27_cov_job_germ3_12
   Job ID:   645802952
   File:     /datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/generated-scripts/s27_cov_job_germ3_12.o645802952
   Updated:  Fri Aug 19 22:30:05 UTC 2016
   Contents Excerpt:
   Error in smooth.spline(gc_sub[, 5], cov_sub[, 5], df = 6) : 
   need at least four unique 'x' values
   Calls: predict -> smooth.spline
   Execution halted
   Error in smooth.spline(gc_sub[, 5], cov_sub[, 5], df = 6) : 
   need at least four unique 'x' values
   Calls: predict -> smooth.spline
   Execution halted
   The method 'do_run' exited abnormally so the Runner will terminate here!
   Return value was: 1
2. A second report from the same process/step (?)
   Job Name: s27_cov_job_germ3_12
   Job ID:   645802952
   File:     /datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/generated-scripts/s27_cov_job_germ3_12.e645802952
   Updated:  Fri Aug 19 22:30:05 UTC 2016
   Contents Excerpt:
   Command run: 
   bash -lc /datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/generated-scripts/s27_cov_job_germ3_12.sh 
   ### gc normalization for coverage track: delly_results/cov/DTB-005_germ_10kb.cov

gc norm compare: FALSE 
[SeqWare Pipeline] ERROR [2016/08/19 22:30:05] | The exit value was 1
Command exit code: 1
### gc normalization for coverage track: delly_results/cov/DTB-005_germ_10kb.cov

gc norm compare: FALSE 
## Full output: /datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/generated-scripts/s27_cov_job_germ3_12.stderr
1. The very bottom of the log file:
   The method 'do_run' exited abnormally so the Runner will terminate here!
   Return value was: 15
   mv: cannot stat `/datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/*.vcf.gz': No such file or directory
   mv: cannot stat`/datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/_.bedpe.txt': No such file or directory
   mv: cannot stat `/datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/delly_results/_.sv.cov.tar.gz': No such file or directory
   mv: cannot stat `/datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/delly_results/*.sv.cov.plots.tar.gz': No such file or directory
   mv: cannot stat`/datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/_.sv.log.tar.gz': No such file or directory
   mv: cannot stat `/datastore/oozie-90b794c1-8645-4a03-9aad-539371d6d9cb/_.json': No such file or directory
   Error while running job: Did not find output file with glob pattern: '['*.somatic.sv.vcf.gz']'
   [job 140372460841680] completed permanentFail
   Final process status is permanentFail
   Workflow error, try again with --debug for more information:
   Process status is ['permanentFail']

22:30:18.756 [main] DEBUG io.dockstore.common.Utilities - stdout was: 

That is all I've got so far.  It looks like all of the proceeding steps prior to this were successful, in that according to the log file the various .vcf files were created and written to the correct directories inside the container--they just never made it out.

Any help or insights would be greatly appreciated.

Thanks,

-- Marc

**Comment by denis-yuen** [comment]
Hi,
This looks like a genuine workflow issue, you may wish to contact @briandoconnor . I think the 

```
Error in smooth.spline(gc_sub[, 5], cov_sub[, 5], df = 6) :
need at least four unique 'x' values
Calls: predict -> smooth.spline
Execution halted
```

 segments mean that even if you got some output, it probably is invalid. 

If you don't get a response from @briandoconnor in a timely fashion, you can also try the staging version of Dockstore with the CWL 1.0 version of this tool. Roughly speaking, you will need to install the versions of cwltool and its dependencies as described here https://staging.dockstore.org/onboarding and then this branch of delly https://github.com/ICGC-TCGA-PanCancer/pcawg_delly_workflow/tree/feature/cwl1

**Comment by mdperry** [comment]
Hi Denis,

Okay, I will try Brian.  I wrote to him roughly a week ago just before I
started this delly workflow, but have not heard back.  I will try again.

Thanks,

-- Marc

**Comment by mdperry** [comment]
I was trying to used this workflow on WXS datasets, and @briandoconnor helped me diagnose the error message, which was that there was not enough depth of coverage somewhere, and so the program was crashing/bailing.

Since then, following up on Brian's suggestion, I obtained a WGS dataset from the PCAWG project (the one that Keiran mentions using as a test case in several of his GitHub repos), and I was able to get this dockstore workflow to run as advertised.

The output files were not identical to the vcf files available through the PCAWG GNOS servers, but that may be because the actual version of delly that was used in production for PCAWG is not the same as the one in this docker image (I think).  The vcf files shared ~ 95% of the called structural variants, and I was happy with that.

N.B. For running delly on WXS I have had success using Tobias' binary files and running them on aws.  It is not as elegantly automated as the docker image, but it works.
